nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/SdkPackageUpdater.cs

```diff
@@ -254,62 +254,133 @@ private static UpdateResult TryUpdateDependencyVersion(ImmutableArray<ProjectBui
 
         foreach (var buildFile in buildFiles)
         {
-            var updateAttributes = new List<XmlAttributeSyntax>();
-            var packageNodes = FindPackageNode(buildFile, dependencyName);
+            var updateNodes = new List<XmlNodeSyntax>();
+            var packageNodes = FindPackageNodes(buildFile, dependencyName);
 
             var previousPackageVersion = previousDependencyVersion;
 
             foreach (var packageNode in packageNodes)
             {
                 var versionAttribute = packageNode.GetAttribute("Version", StringComparison.OrdinalIgnoreCase)
                     ?? packageNode.GetAttribute("VersionOverride", StringComparison.OrdinalIgnoreCase);
-                if (versionAttribute is null)
+                var versionElement = packageNode.Elements.FirstOrDefault(e => e.Name.Equals("Version", StringComparison.OrdinalIgnoreCase))
+                    ?? packageNode.Elements.FirstOrDefault(e => e.Name.Equals("VersionOverride", StringComparison.OrdinalIgnoreCase));
+                if (versionAttribute is not null)
                 {
-                    continue;
-                }
+                    // Is this the case where version is specified with property substitution?
+                    if (versionAttribute.Value.StartsWith("$(") && versionAttribute.Value.EndsWith(")"))
+                    {
+                        propertyNames.Add(versionAttribute.Value.Substring(2, versionAttribute.Value.Length - 3));
+                    }
+                    // Is this the case that the version is specified directly in the package node?
+                    else
+                    {
+                        var currentVersion = versionAttribute.Value.TrimStart('[', '(').TrimEnd(']', ')');
+                        if (currentVersion.Contains(',') || currentVersion.Contains('*'))
+                        {
+                            logger.Log($"    Found unsupported [{packageNode.Name}] version attribute value [{versionAttribute.Value}] in [{buildFile.RepoRelativePath}].");
```

Comment:
q: can we not just use a `\t` here instead? It feels weird to just have a bunch of spaces at the beginning of the string.

Reply:
I agree (I'd actually go farther and say that log indentation/prefix should be managed by BeginScope or a similar concept) but right now I'm just doing what matches the rest of the file. If we wanted to revisit how the NuGetUpdater handles logging as a separate PR/set of PRs I'd be totally down with that.